### ISSUE TITLE
sui-search does not show results with Angular 6+

### ISSUE THREAD

**Issue by christopher-harris** [edcarroll/ng2-semantic-ui#416]
Please read https://github.com/edcarroll/ng2-semantic-ui/blob/master/CONTRIBUTING.md and search
existing issues (both open and closed) prior to opening any new issue and ensure you follow the instructions therein.

### Bug description:
Upgraded to Angular 6.1.9 and updated ng2-semantic-ui to latest version. When typing in a sui-search component, the results are not being displayed in the browser. Looking at the DOM in DevTools, I am seeing the results element, but it looks like the classes for the input tag and the results div are not getting updated. If I manually update the classes to what they are suppose to be, I see the results as expected. If I click on the input again, I see the results as expected.

### Link to minimally-working plunker that reproduces the issue:
Could not get plunker to work, so I used StackBlitz
https://stackblitz.com/edit/angular-pdgnzh

### Version of Angular, ng2-semantic-ui, and Semantic UI:

Angular: 6.1.9

ng2-semantic-ui: 0.9.7

Semantic UI: 2.4.0

**Comment by thardy** [comment]
I'm seeing the same behavior. A sui-search that was working fine in Angular 4.x has now stopped displaying results after upgrading to Angular 6.

**Comment by dpfavand** [comment]
I'm also experiencing this. I've narrowed it down to the following:

* If the search input receives focus but there is no text in the search input, it will not show the result list, even once text is entered.
* If there is text already in the search input, it will show the result list, even if all text is removed.

**Comment by preethis610** [comment]
Yes I am also facing the same issue..Along with this I am facing the following issue too.**I am getting the search result as "No results"** eventhough the search value exists.**Semantic search is not calling optionsLookup for already executed search string.**
_html:_
**<sui-search [optionsLookup]="searchCall" [optionsField]="'name'" #suiSearch ></sui-search>**  
_ts:_
 **searchCall=(query)=>{
		    return this.service.apiCall(query);
			}**** 
If user searches for a query say "**Apple1 Fruit**" for the first time, optionsLookup executes and shows the result correctly."Apple2 Fruit" the next query also does optionsLookup and shows the result.If the already executed query is searched again by the user (change 2 to 1 in the search string-"**Apple1 Fruit**"),optionsLookup  is not called.Semantic search shows the result from its past search results that has it, which fails most of the time(getting No results).

How semantic is displaying the result sometimes without optionLookup?Is there any way to call optionsLookup always, so that result is always from an api.

Version of Angular: 6.0.5
Version of ng2-semantic-ui: 0.9.7

**Comment by NikitaObukhov** [comment]
I had same issue. This is bug in typescript compiler and it's is just hilarious.

This is original Typescript file ./src/modules/search/components/search.ts:
```javascript   
 public set query(query:string) {
        this.selectedResult = undefined;
        // Initialise a delayed search.
        this.searchService.updateQueryDelayed(query, () =>
            // Set the results open state depending on whether a query has been entered.
            this.dropdownService.setOpenState(this.searchService.query.length > 0));
    }
```

And it is compiled into:
./dist/modules/search/components/search.js:
```javascript
    Object.defineProperty(SuiSearch.prototype, "query", {
        get: function () {
            return this.searchService.query;
        },
        set: function (query) {
            var _this = this;
            this.selectedResult = undefined;
            // Initialise a delayed search.
            this.searchService.updateQueryDelayed(query, function () {
                // Set the results open state depending on whether a query has been entered.
                return
                // Set the results open state depending on whether a query has been entered.
                _this.dropdownService.setOpenState(_this.searchService.query.length > 0);
            });
        },
        enumerable: true,
        configurable: true
    });
```
(Compile command is: node ./scripts/clean && tsc -p src/tsconfig.json && ngc -p src/tsconfig.json)

Bug is here:
```javascript  
              return
                // Set the results open state depending on whether a query has been entered.
                _this.dropdownService.setOpenState(_this.searchService.query.length > 0);
```

Function just returns nothing and two next lines are not executed (God bless Javascript, it does not require semicolons).

So this bug was probably introduced in this commit: https://github.com/edcarroll/ng2-semantic-ui/commit/6f14e2efb982e778d37aa84352b5e6342d15b554

because if you remove comment line from Typescript file (-'// Set the results open state depending on whether a query has been entered.') everything compiles good:

Removed comment line from typescript:
```javascript
public set query(query:string) {
        this.selectedResult = undefined;
        // Initialise a delayed search.
        this.searchService.updateQueryDelayed(query, () =>
            this.dropdownService.setOpenState(this.searchService.query.length > 0));
    }
```

Compiled without bug:
```javascript
    Object.defineProperty(SuiSearch.prototype, "query", {
        get: function () {
            return this.searchService.query;
        },
        set: function (query) {
            var _this = this;
            this.selectedResult = undefined;
            // Initialise a delayed search.
            this.searchService.updateQueryDelayed(query, function () {
                return _this.dropdownService.setOpenState(_this.searchService.query.length > 0);
            });
        },
        enumerable: true,
        configurable: true
    });
```
